x/concentrated-liquidity/lp.go

```diff
@@ -33,42 +30,42 @@ func (k Keeper) createPosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddr
 		return sdk.Int{}, sdk.Int{}, sdk.Dec{}, fmt.Errorf("invalid upper tick: %d", upperTick)
 	}
 
-	if liquidityIn.IsZero() {
+	// now calculate amount for token0 and token1
+	pool := k.getPoolbyId(ctx, poolId)
+
+	currentSqrtPrice := pool.CurrentSqrtPrice
+	sqrtRatioUpperTick, _ := k.tickToSqrtPrice(sdk.NewInt(upperTick))
+	sqrtRatioLowerTick, _ := k.tickToSqrtPrice(sdk.NewInt(lowerTick))
+
+	liquidity := getLiquidityFromAmounts(currentSqrtPrice, sqrtRatioUpperTick, sqrtRatioLowerTick, amount0Desired, amount1Desired)
+	if liquidity.IsZero() {
 		return sdk.Int{}, sdk.Int{}, sdk.Dec{}, fmt.Errorf("token in amount is zero")
 	}
 
 	// update tickInfo state
 	// TODO: come back to sdk.Int vs sdk.Dec state & truncation
-	err = k.initOrUpdateTick(ctx, poolId, lowerTick, liquidityIn.TruncateInt(), false)
+	err = k.initOrUpdateTick(ctx, poolId, lowerTick, liquidity.TruncateInt(), false)
 	if err != nil {
 		return sdk.Int{}, sdk.Int{}, sdk.Dec{}, err
 	}
 
 	// TODO: come back to sdk.Int vs sdk.Dec state & truncation
-	err = k.initOrUpdateTick(ctx, poolId, upperTick, liquidityIn.TruncateInt(), true)
+	err = k.initOrUpdateTick(ctx, poolId, upperTick, liquidity.TruncateInt(), true)
 	if err != nil {
 		return sdk.Int{}, sdk.Int{}, sdk.Dec{}, err
 	}
 
 	// update position state
 	// TODO: come back to sdk.Int vs sdk.Dec state & truncation
-	err = k.initOrUpdatePosition(ctx, poolId, owner, lowerTick, upperTick, liquidityIn.TruncateInt())
+	err = k.initOrUpdatePosition(ctx, poolId, owner, lowerTick, upperTick, liquidity.TruncateInt())
 	if err != nil {
 		return sdk.Int{}, sdk.Int{}, sdk.Dec{}, err
 	}
 
-	// now calculate amount for token0 and token1
-	pool := k.getPoolbyId(ctx, poolId)
-
-	currentSqrtPrice := pool.CurrentSqrtPrice
-	sqrtRatioUpperTick, _ := k.tickToSqrtPrice(sdk.NewInt(upperTick))
-	sqrtRatioLowerTick, _ := k.tickToSqrtPrice(sdk.NewInt(lowerTick))
-
-	amtDenom0 = calcAmount0Delta(liquidityIn, currentSqrtPrice, sqrtRatioUpperTick).RoundInt()
-	amtDenom1 = calcAmount1Delta(liquidityIn, currentSqrtPrice, sqrtRatioLowerTick).RoundInt()
+	amtDenom0 = calcAmount0Delta(liquidity, currentSqrtPrice, sqrtRatioUpperTick).RoundInt()
+	amtDenom1 = calcAmount1Delta(liquidity, currentSqrtPrice, sqrtRatioLowerTick).RoundInt()
 
-	// TODO: calculate from final amounts
-	liquidityCreated = sdk.MustNewDecFromStr("1517.882323")
+	liquidityCreated = getLiquidityFromAmounts(currentSqrtPrice, sqrtRatioUpperTick, sqrtRatioLowerTick, amtDenom0, amtDenom1)
 
 	return amtDenom0, amtDenom1, liquidityCreated, nil
```

Comment:
should we have it not return liquidityCreated any more?

Reply:
as discussed offline, yeah this can be removed